lib/dom.mli

```diff
@@ -1503,9 +1503,16 @@ module Style : sig
   let rubyPosition = string_style_prop "rubyPosition"]
 end
 
-module DangerouslySetInnerHTML : sig
+module SafeString : sig
   type t
-end
 
-val makeInnerHtml : __html:string -> DangerouslySetInnerHTML.t
-  [@@js.builder] [@@js.verbatim_names]
+  val make_unchecked : string -> t
+    [@@js.custom
+      val make_react_html_object : __html:string -> t
+        [@@js.builder] [@@js.verbatim_names]
```

Comment:
I think `t` ought to be a `private string`, and have `dangerouslySetInnerHTML` create this object instead. That would allow this type to be more generally useful. It's also more idiomatic OCaml.

Reply:
Having `dangerouslySetInnerHTML` create the `__html` object makes sense.

I moved some code in https://github.com/ml-in-barcelona/jsoo-react/pull/128/commits/d3924adb125a8a7fadb8c2535486127c9b05c1fa, let me know if this is somehow what you had in mind. As we have now `Dom` and `Dom_html` and it's unclear to me what should go where 😄

Reply:
LGTM. Thanks!